nets/chain_spec/addAuthorities.ts

```diff
@@ -1,24 +1,31 @@
 import { GenesisConfig } from "./ChainSpec.ts"
 
-// TODO: #889 add support for pallet_session, pallet_aura and pallet_grandpa
-
 export function addAuthorities(genesisConfig: GenesisConfig, count: number) {
   if (count > authorities.length) {
     throw new Error(`authorities count should be <= ${authorities.length}`)
   }
-  if (!genesisConfig.session) return
-  genesisConfig.session.keys.length = 0
-  authorities.slice(0, count).forEach(({ srAccount, srStash, edAccount, ecAccount }) =>
-    genesisConfig.session!.keys.push([srStash, srStash, {
-      grandpa: edAccount,
-      babe: srAccount,
-      im_online: srAccount,
-      para_validator: srAccount,
-      para_assignment: srAccount,
-      authority_discovery: srAccount,
-      beefy: ecAccount,
-    }])
-  )
+  if (genesisConfig.session) {
+    genesisConfig.session.keys = authorities.slice(0, count).map((
+      { srAccount, srStash, edAccount, ecAccount },
+    ) => [
+      srStash,
+      srStash,
+      {
+        grandpa: edAccount,
+        babe: srAccount,
+        im_online: srAccount,
+        para_validator: srAccount,
+        para_assignment: srAccount,
+        authority_discovery: srAccount,
+        beefy: ecAccount,
+      },
+    ])
+  } else if (genesisConfig.aura && genesisConfig.grandpa) {
+    genesisConfig.aura.authorities = authorities.slice(0, count).map(({ srAccount }) => srAccount)
```

Comment:
Let's extract `authorities.slice(0, count)` to a preceding local & reuse it in the following two lines.